src/Model/Operation/ProductSyncHandler.php

```diff
@@ -333,19 +342,39 @@ private function handleFirstActiveVariant(ProductEntity $product): ?ProductEntit
         return $mainProduct;
     }
 
+    private function handleFirstVariantInStock(
+        ProductEntity $product,
+        SalesChannelContext $context,
+    ): ?ProductEntity
+    {
+        $mainProduct = null;
+        $variants = new ProductCollection([$product]);
+
+        foreach ($product->getChildren() as $child) {
+            $stock = $this->getProductStock($product, $context);
+
+            if ($child->getActive() && !$mainProduct && ($stock || !$child->getIsCloseout())) {
+                $mainProduct = $child;
+            } else {
+                $variants->add($child);
+            }
+        }
+
+        if ($mainProduct) {
+            $mainProduct->setChildren($variants);
+        }
+
+        return $mainProduct;
```

Comment:
So by this logic:
1. We have a product which has stock and has skus -> we send the product and skus to Nosto
2. If out of stock and clearance we set SKU as a product and set skus for that -> we send the product and skus to Nosto
In this case what happens with the old product in Nosto? with the skus

Reply:
The old product will be discontinued, meaning its status will be set to "Discontinued".